src/ParallelAlgorithms/Events/ObserveFields.hpp

```diff
@@ -235,24 +235,44 @@ class ObserveFields<VolumeDim, tmpl::list<Tensors...>,
         0_st));
 
     const auto record_tensor_component_impl =
-        [&components, &interpolant](const auto& tensor,
-                                    const FloatingPointType floating_point_type,
-                                    const std::string& tag_name) {
+        [&components](DataVector&& tensor_component,
+                      const FloatingPointType floating_point_type,
+                      const std::string& component_name) {
+          if (floating_point_type == FloatingPointType::Float) {
+            components.emplace_back(component_name,
+                                    std::vector<float>{tensor_component.begin(),
+                                                       tensor_component.end()});
+          } else {
+            components.emplace_back(component_name,
+                                    std::move(tensor_component));
+          }
+        };
+
+    const auto record_tensor_components_impl =
+        [&record_tensor_component_impl, &interpolant](
+            const auto& tensor, const FloatingPointType floating_point_type,
```

Comment:
This is not meant to be a late review or anything.

Just an FYI, in C++20 you can now do
```cpp
[&record_tensor_component_impl, &interpolant]<typename TensorType>(const TensorType&, ...)
```
to get template parameters in lambdas. You could also do
```cpp
[&record_tensor_component_impl, &interpolant]<typename X, typename Y, typename Z>(const Tensor<X,Y,Z>&, ...)
```
with better names than X,Y,Z :)